### PR TITLE
MINIFI-12 initial commit of http config change notifier

### PR DESCRIPTION
**Pull request by JPercivall**
Created an http notifier that uses Jetty to listen for requests

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

**Comment by apiri** [comment]
There were a few checkstyle errors for unused imports:

> [WARNING] src/main/java/org/apache/nifi/minifi/bootstrap/configuration/RestChangeNotifier.java[20:8] (imports) UnusedImports: Unused import - org.eclipse.jetty.server.ConnectionFactory.
[WARNING] src/main/java/org/apache/nifi/minifi/bootstrap/configuration/RestChangeNotifier.java[24:8] (imports) UnusedImports: Unused import - org.eclipse.jetty.server.SslConnectionFactory.
[WARNING] src/main/java/org/apache/nifi/minifi/bootstrap/configuration/RestChangeNotifier.java[40:8] (imports) UnusedImports: Unused import - java.util.Collection.

**Comment by JPercivall** [comment]
Well doh, looks like I forgot to run a contrib check after I squashed some commits and they snuck by. I just posted a new commit.

**Comment by apiri** [comment]
no worries.  adjusted here and kept on with review.

File: minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/src/main/resources/conf/bootstrap.conf [hunk 1]
```diff
@@ -41,6 +42,11 @@ nifi.minifi.config=./conf/config.yml
 # How frequently the file specified by 'nifi.minifi.notifier.file.config.path' should be evaluated for changes.
 #nifi.minifi.notifier.file.polling.period.seconds=5
 
+# Rest change notifier configuration
+
+# Port on which the Jetty server will bind to, keep commented for a random open port
+#nifi.minifi.notifier.http.port=8338
+
```
Comment (apiri) [inline minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/src/main/resources/conf/bootstrap.conf]: do you think it makes sense to include the stubbed out options for the mutual auth notifier configuration as well?

Comment (JPercivall) [inline minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/src/main/resources/conf/bootstrap.conf]: I could add it but my original thinking was just to add the basics needed to configure the notifier. I thought that was better than to clutter up the conf file with a bunch of commented out unused lines.

File: pom.xml [hunk 2]
```diff
@@ -177,6 +178,16 @@ limitations under the License.
...
                 <scope>provided</scope>
             </dependency>
+            <dependency>
+                <groupId>org.eclipse.jetty</groupId>
+                <artifactId>jetty-server</artifactId>
+                <version>${jetty.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.eclipse.jetty</groupId>
+                <artifactId>jetty-servlet</artifactId>
```
Comment (apiri) [inline pom.xml]: Licensing looks okay for Jetty as it is dual licensed EPL & ASLv2, however there is a NOTICE at https://github.com/eclipse/jetty.project/blob/master/NOTICE.txt where I think we need to carry over the copyright in our NOTICE.

File: minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/src/main/resources/conf/bootstrap.conf (hunk 1 above)
Comment (apiri) [inline minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/src/main/resources/conf/bootstrap.conf]: Okay, works for me.

File: pom.xml (hunk 2 above)
Comment (JPercivall) [inline pom.xml]: Yup I will add the same Jetty section that NiFi uses:

  (ASLv2) Jetty
    The following NOTICE information applies:
       Jetty Web Container
       Copyright 1995-2015 Mort Bay Consulting Pty Ltd.

**Comment by apiri** [comment]
Overall looks good.  Was able to verify intended functionality with both open and mutual auth configurations.  Once the minor comments are addressed, a +1 from me.  Thanks, Joe.

**Comment by JPercivall** [comment]
Awesome, thanks Aldrin. I will add the NOTICE information for jetty and merge it in.

**Pull request closed without merging**